registry/auth.go

```diff
@@ -92,21 +92,23 @@ func (auth *RequestAuthorization) Authorize(req *http.Request) error {
 }
 
 // Login tries to register/login to the registry server.
-func Login(authConfig *cliconfig.AuthConfig, registryEndpoint *Endpoint, factory *requestdecorator.RequestFactory) (string, error) {
+func Login(authConfig *cliconfig.AuthConfig, registryEndpoint *Endpoint) (string, error) {
+	client := registryEndpoint.HTTPClient()
+	client.Transport = transport.NewTransport(client.Transport, DockerHeaders(nil)...)
```

Comment:
`DockerHeaders` replaces the `HTTPRequestFactory` from the only caller https://github.com/docker/docker/pull/13265/files#diff-2b07b8a4e93be324f246494ec743cde2L35